poetry/core/poetry.py

```diff
@@ -37,5 +40,15 @@ def package(self):  # type: () -> "ProjectPackage"
     def local_config(self):  # type: () -> dict
         return self._local_config
 
+    @property
+    def locker(self):  # type: () -> Optional["Locker"]
+        if self._locker is None:
+            from poetry.core.lock.locker import Locker
+
+            self._locker = Locker(
+                self.pyproject.file / "poetry.lock", self.local_config
```

Comment:
Seems to me like it should be `self.pyproject.file.path.parent / "poetry.lock"`.